Generate JAR files for the library

**Description**
Although we have the `ios` and `android` module set up, we need to expect the library to generate files as well for use on desktop JAVA since the AAR files cannot be understood by then. 

**Solution**
We may need another module set up similar to `ios` or `android` that can be outputted as `jar`. 

**Alternative**
N/A

@barbeau I was able to publish the library to the local maven repository and log the data in the project! Here's the link to the test repository: https://github.com/BumbleFlash/opentripplanner-multiplatform-test and the updated library code has been pushed to #13. 

One catch is, when I run the test application using `gradlew build`, the exceptions mentioned below are thrown. 

```
C:\Users\sudarshans\IdeaProjects\opentripplanner-multiplatform-test>gradlew build

BUILD SUCCESSFUL in 3s
3 actionable tasks: 1 executed, 2 up-to-date
C:\Users\sudarshans\IdeaProjects\opentripplanner-multiplatform-test>java -jar build/libs/opentripplanner-multiplatform-test-1.0-SNAPSHOT.jar
Exception in thread "main" java.lang.NoClassDefFoundError: kotlinx/coroutines/CoroutineScope
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at MainKt.main(Main.kt:2)
        at MainKt.main(Main.kt)
Caused by: java.lang.ClassNotFoundException: kotlinx.coroutines.CoroutineScope
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 11 more

```
The CoroutineScope is used by the desktop application and not by the library. 

However, I have added the configuration to generate a `fatJar` file. To build it, type `gradle fatJar` on the command line and run the jar file generated in `build/libs`. It seems to be working fine this way.

--- a/README.md
+++ b/README.md
@@ -92,7 +92,7 @@ TODO
 
 #### Desktop Java/Kotlin
 
-This command generates 4 Jar files in `library/build/libs` and we encourage to use `library-jvm-1.0.0.jar` for desktop applications.
+This command generates four JAR files in `library/build/libs` and we encourage to use `library-jvm-1.0.0.jar` for manually including dependencies:
 
 ```
 ./gradlew :library:build

--- a/README.md
+++ b/README.md
@@ -92,7 +92,7 @@ TODO
 
 #### Desktop Java/Kotlin
 
-This command generates four JAR files in `library/build/libs` and we encourage to use `library-jvm-1.0.0.jar` for manually including dependencies:
+This command generates four JAR files in `library/build/libs` and we encourage you to use `library-jvm-1.0.0.jar` for manually including dependencies:
 
 ```
 ./gradlew :library:build

--- a/README.md
+++ b/README.md
@@ -92,7 +92,11 @@ TODO
 
 #### Desktop Java/Kotlin
 
-TODO
+This command generates 4 Jar files in `library/build/libs` and we encourage to use `library-jvm-1.0.0.jar` for desktop applications.
+
+```
+./gradlew :library:build
+```
 
 ## License
 ```

--- a/library/build.gradle.kts
+++ b/library/build.gradle.kts
@@ -14,7 +14,11 @@ version = "1.0"
 
 kotlin {
     android()
-    jvm()
+    jvm {
+        compilations.all {
+            kotlinOptions.jvmTarget = "1.8"
+        }
+    }
     ios {
         binaries {
             framework {

--- a/library/build.gradle.kts
+++ b/library/build.gradle.kts
@@ -9,6 +9,9 @@ plugins {
     id("kotlin-android-extensions")
 }
 
+group = "edu.usf.cutr.otp"
+version = "1.0"
+
 kotlin {
     android()
     jvm()
@@ -47,7 +50,7 @@ kotlin {
 
         val jvmMain by getting {
             dependencies {
-                implementation("io.ktor:ktor-client-apache:$ktorVersion")
+                implementation("io.ktor:ktor-client-java:$ktorVersion")
             }
         }
     }

--- a/library/build.gradle.kts
+++ b/library/build.gradle.kts
@@ -11,6 +11,7 @@ plugins {
 
 kotlin {
     android()
+    jvm()
     ios {
         binaries {
             framework {
@@ -41,6 +42,12 @@ kotlin {
         val iosMain by getting {
             dependencies {
                 implementation("io.ktor:ktor-client-ios:$ktorVersion")
+            }
+        }
+
+        val jvmMain by getting {
+            dependencies {
+                implementation("io.ktor:ktor-client-apache:$ktorVersion")
             }
         }
     }

--- a/library/build.gradle.kts
+++ b/library/build.gradle.kts
@@ -11,7 +11,7 @@ plugins {
 }
 
 group = "edu.usf.cutr.otp"
-version = "1.0.2"
+version = "1.0.0"
 
 repositories {
     mavenLocal()

--- a/library/build.gradle.kts
+++ b/library/build.gradle.kts
@@ -7,10 +7,15 @@ plugins {
     id("com.android.library")
     id("kotlinx-serialization")
     id("kotlin-android-extensions")
+    id("maven-publish")
 }
 
 group = "edu.usf.cutr.otp"
-version = "1.0"
+version = "1.0.2"
+
+repositories {
+    mavenLocal()
+}
 
 kotlin {
     android()
@@ -55,6 +60,7 @@ kotlin {
         val jvmMain by getting {
             dependencies {
                 implementation("io.ktor:ktor-client-java:$ktorVersion")
+                implementation ("org.jetbrains.kotlinx:kotlinx-coroutines-swing:1.4.2")
             }
         }
     }

--- a/library/src/jvmMain/kotlin/edu/usf/cutr/otp/Dispatcher.kt
+++ b/library/src/jvmMain/kotlin/edu/usf/cutr/otp/Dispatcher.kt
@@ -1,0 +1,6 @@
+package edu.usf.cutr.otp
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+
+internal actual val ApplicationDispatcher: CoroutineDispatcher = Dispatchers.Default

